openfeature-provider/go/local_resolver_factory.go

```
@@ -0,0 +1,295 @@
+			initialState = []byte{}
+		}
+
+		// Create flag logger based on disableLogging flag
+		if disableLogging {
+			flagLogger = NewNoOpWasmFlagLogger()
+		} else {
+			// If logging is enabled but using StateProvider, use NoOp for now
+			// Users can extend this to provide their own logger
+			flagLogger = NewNoOpWasmFlagLogger()
+		}
+	} else {
+		// Use gRPC-based state fetcher and flag logger
+		// Create TLS credentials for secure connections
+		tlsCreds := credentials.NewTLS(nil)
+
+		// Create auth service connection (no auth interceptor for this one)
+		authConn, err := grpc.NewClient(authServiceAddr, grpc.WithTransportCredentials(tlsCreds))
+		if err != nil {
+			return nil, err
+		}
+		authService := iamv1.NewAuthServiceClient(authConn)
+
+		// Create token holder
+		tokenHolder := NewTokenHolder(apiClientID, apiClientSecret, authService)
+
+		// Create JWT auth interceptor
+		authInterceptor := NewJwtAuthInterceptor(tokenHolder)
+
+		// Create gRPC connection for resolver state service with auth
+		stateConn, err := grpc.NewClient(
+			resolverStateServiceAddr,
+			grpc.WithTransportCredentials(tlsCreds),
+			grpc.WithUnaryInterceptor(authInterceptor.UnaryClientInterceptor()),
+			grpc.WithStreamInterceptor(authInterceptor.StreamClientInterceptor()),
+		)
+		if err != nil {
+			return nil, err
+		}
+		resolverStateService := adminv1.NewResolverStateServiceClient(stateConn)
+
+		// Get account name from token
+		token, err := tokenHolder.GetToken(ctx)
+		if err != nil {
+			log.Printf("Warning: failed to get initial token, account name will be unknown: %v", err)
+		}
+		accountName := "unknown"
+		if token != nil {
+			accountName = token.Account
+		}
+
+		// Create state fetcher
+		stateFetcher = NewFlagsAdminStateFetcher(resolverStateService, accountName)
+
+		// Do initial state fetch to get initial state for SwapWasmResolverApi
+		if err := stateFetcher.Reload(ctx); err != nil {
+			log.Printf("Initial state fetch failed, using empty state: %v", err)
```

should throw an error?